### PR TITLE
RFC#50-P1 Restructuring some routes and adding a feature flag

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This is part of a [bigger endeavour](https://github.com/forem/rfcs/pull/50) to restructure admin routes and navigation. 

In this PR, I set the foundation by grouping the routes under the `:admin` namespace as follows; 
- At the top I've grouped the routes that will most likely remain as is  (this may change as I start exploring more)   
- Thereafter, I've grouped existing routes that will be replaced by new ones. I've added a comment to indicate that I will be deleting those routes
- Finally, I've grouped the newly scoped routes under a feature flag "admin_restructure". You can find a table of these new routes [here](https://github.com/forem/rfcs/blob/aefc141f386b395cc12d4b9f56816f1de4702eab/text/0050-update-admin-ia-nav.md#3-move-existing-content-into-new-location)
 
Please note that the new routes are not meant to be fully functional or 100% working as yet, and since they are under a feature flag, this is okay. I will go through each route in another PR and ensure that we provide updated links etc. that will not break the current admin interface. 

## Related Tickets & Documents
https://github.com/forem/rfcs/pull/50

## QA Instructions, Screenshots, Recordings
I think that the only testing that needs to occur now is to ensure that the feature flag is working.

- Open up a _new_ route `http://localhost:3000/admin/customization/html_variants`. You should get a no route matches error.
- Add a feature flag called `admin_restructure` to http://localhost:3000/admin/feature_flags, and enable it.
- Now open up `http://localhost:3000/admin/customization/html_variants` and you should see the html_variants page

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

I'm not quite sure if this needs to be tested.

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: It will be communicated once its complete.

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -71,89 +83,155 @@
...
       # NOTE: @citizen428 The next two resources have a temporary constraint
       # while profile generalization is still WIP
       constraints(->(_request) { FeatureFlag.enabled?(:profile_admin) }) do
         resources :profile_field_groups, only: %i[update create destroy]
         resources :profile_fields, only: %i[index update create destroy]
       end
-      resources :reactions, only: [:update]
-      resources :response_templates, only: %i[index new edit create update destroy]
+
+      # These redirects serve as a safegaurd to prevent 404s for any Admins
```
Comment (Ridhwana) [inline config/routes.rb:94]: I'm not sure was there an agreement for how long should these stay in? Are they forever routes?

File: config/routes.rb [hunk 2]
```diff
@@ -71,89 +83,155 @@
...
-      end
-      resources :organization_memberships, only: %i[update destroy create]
-      resources :organizations, only: %i[index show] do
-        member do
-          patch "update_org_credits"
-        end
-      end
-      resources :sponsorships, only: %i[index edit update new create destroy]
-      resources :welcome, only: %i[index create]
-      resources :growth, only: %i[index]
```
Comment (Ridhwana) [inline config/routes.rb:136]: Removed the growth route as it was no longer being used anywhere

File: config/routes.rb (hunk 1 above)
Comment (rhymes) [reply config/routes.rb:94]: Not sure, being admin routes I think we can eventually remove them though. 

ps. tiny typo in `safeguard`

**Review by rhymes — COMMENTED** [review]
One thing: can you add the new feature flag programmatically? See https://docs.forem.com/technical-overview/feature_flags/#creating-a-new-feature-flag

This way it'll be available on each Forem and we can remove it when we are done without leaving behind obsolete data in the DB

**Comment by Ridhwana** [comment]
> One thing: can you add the new feature flag programmatically? See https://docs.forem.com/technical-overview/feature_flags/#creating-a-new-feature-flag
> 
> This way it'll be available on each Forem and we can remove it when we are done without leaving behind obsolete data in the DB


@rhymes I did think about this because we did it for Data Update Scripts, but in this case I'm not sure that it makes sense to add this feature flag to be available on every Forem right now. Currently, it's a work in progress where the links within the pages still redirect to the original routes etc. so seeing and switching this feature flag on any Forem would confuse admins so much if they come across it. I feel like there will be a stage at which we want to test this on other Forems and at that point I'll add the DUS.

File: config/routes.rb (hunk 1 above)
Comment (Ridhwana) [reply config/routes.rb]: lol oops 😅

**Comment by rhymes** [comment]
@Ridhwana makes sense, let's keep it a "secret" 🕵️

**Comment by citizen428** [comment]
> I'm not quite sure if this needs to be tested.

You could throw in a [routing spec](https://relishapp.com/rspec/rspec-rails/docs/routing-specs) for a handful of the routes and test with the feature flag enabled and disabled.

**Review by citizen428 — APPROVED** [review]
You could use a similar approach as we used in forem-app here to organize the routes in different files (after we upgrade to Rails 6.1 we can use them as the basis for separate routes files to use with the new `draw` route), but that's probably not worth the effort.

**Review by rhymes — APPROVED** [review]
LGTM!

**Pull request merged**